pulp_smash/tests/platform/api_v2/test_repository.py

```diff
@@ -0,0 +1,235 @@
+    valid.
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create three repositories to read, update, and delete."""
+        cls.cfg = get_config()
+        cls.update_body = {
+            'delta': {
+                'display_name': rand_str(),
+                'description': rand_str()
+            }
+        }
+        cls.bodies = [{'id': rand_str()} for _ in range(3)]
+        cls.paths = []
+        for body in cls.bodies:
+            response = requests.post(
+                cls.cfg.base_url + REPOSITORY_PATH,
+                json=body,
+                **cls.cfg.get_requests_kwargs()
+            )
+            response.raise_for_status()
+            cls.paths.append(response.json()['_href'])
+
+        # Read, update, and delete the three repositories, respectively.
+        cls.read_response = requests.get(
+            cls.cfg.base_url + cls.paths[0],
+            **cls.cfg.get_requests_kwargs()
+        )
+        cls.update_response = requests.put(
+            cls.cfg.base_url + cls.paths[1],
+            json=cls.update_body,
+            **cls.cfg.get_requests_kwargs()
+        )
+        cls.delete_response = requests.delete(
+            cls.cfg.base_url + cls.paths[2],
+            **cls.cfg.get_requests_kwargs()
+        )
+
+    def test_status_code(self):
+        """Assert that each response has a 200 status code."""
+        expected_status_codes = zip(
+            ('read_response', 'update_response', 'delete_response'),
+            (200, 200, 202)
+        )
+        for attr, expected_status in expected_status_codes:
+            with self.subTest(attr):
+                self.assertEqual(
+                    getattr(self, attr).status_code,
+                    expected_status
+                )
+
+    def test_read_attributes(self):
+        """Assert that the read repository has the correct attributes."""
+        attributes = self.read_response.json()
+        self.assertLessEqual(
+            set(self.bodies[0].keys()),
+            set(attributes.keys())
+        )
+        attributes = {key: attributes[key] for key in self.bodies[0].keys()}
+        self.assertEqual(self.bodies[0], attributes)
+
+    def test_update_attributes_spawned_tasks(self):
+        """Assert that `spawned_tasks` is present and no tasks were created."""
+        response = self.update_response.json()
+        self.assertIn('spawned_tasks', response)
+        self.assertListEqual([], response['spawned_tasks'])
+
+    def test_update_attributes_result(self):
+        """Assert that `result` is present and has the correct attributes."""
+        response = self.update_response.json()
+        self.assertIn('result', response)
+        for key in self.update_body['delta']:
+            with self.subTest(key):
+                self.assertIn(key, response['result'])
+                self.assertEqual(
+                    self.update_body[key],
```

Comment:
Nice catch. This could also be written as:

```python
for key, value in self.update_body['delta'].items():
    with self.subTest(key=key):
        self.assertIn(key, response['result'])
        self.assertEqual(response['result'][key], value)
```